orca-front50/src/main/java/com/netflix/spinnaker/orca/front50/tasks/DeletePipelineTask.java

```diff
@@ -0,0 +1,172 @@
+      ObjectMapper objectMapper) {
+    this.front50Service = front50Service.orElse(null);
+    this.pipelineModelMutators = pipelineModelMutators.orElse(new ArrayList<>());
+    this.objectMapper = objectMapper;
+  }
+
+  private final Front50Service front50Service;
+  private final List<PipelineModelMutator> pipelineModelMutators;
+  ObjectMapper objectMapper;
+
+  @Override
+  public TaskResult execute(StageExecution stage) {
+    if (front50Service == null) {
+      throw new UnsupportedOperationException(
+          "Front50 is not enabled, no way to delete pipeline. Fix this by setting front50.enabled: true");
+    }
+
+    if (!stage.getContext().containsKey("pipeline")) {
+      throw new IllegalArgumentException("pipeline context must be provided");
+    }
+
+    Map<String, Object> pipeline;
+    if (!(stage.getContext().get("pipeline") instanceof String)) {
+      pipeline = (Map<String, Object>) stage.getContext().get("pipeline");
+    } else {
+      pipeline = (Map<String, Object>) stage.decodeBase64("/pipeline", Map.class);
+    }
+
+    if (!pipeline.containsKey("index")) {
+      Map<String, Object> existingPipeline = fetchExistingPipeline(pipeline);
+      if (existingPipeline != null) {
+        pipeline.put("index", existingPipeline.get("index"));
+      }
+    }
+    String serviceAccount = (String) stage.getContext().get("pipeline.serviceAccount");
+    if (serviceAccount != null) {
+      updateServiceAccount(pipeline, serviceAccount);
+    }
+    final Boolean isSavingMultiplePipelines =
+        (Boolean)
+            Optional.ofNullable(stage.getContext().get("isSavingMultiplePipelines")).orElse(false);
+    final Boolean staleCheck =
+        (Boolean) Optional.ofNullable(stage.getContext().get("staleCheck")).orElse(false);
+    if (stage.getContext().get("pipeline.id") != null
+        && pipeline.get("id") == null
+        && !isSavingMultiplePipelines) {
+      pipeline.put("id", stage.getContext().get("pipeline.id"));
+
+      // We need to tell front50 to regenerate cron trigger id's
+      pipeline.put("regenerateCronTriggerIds", true);
+    }
+
+    pipelineModelMutators.stream()
+        .filter(m -> m.supports(pipeline))
+        .forEach(m -> m.mutate(pipeline));
+
+    Response response =
+        front50Service.deletePipelineConfig(
+            pipeline.get("application").toString(), pipeline.get("name").toString(), staleCheck);
+
+    Map<String, Object> outputs = new HashMap<>();
+    outputs.put("notification.type", "deletepipeline");
+    outputs.put("application", pipeline.get("application"));
+    outputs.put("pipeline.name", pipeline.get("name"));
+
+    try {
+      Map<String, Object> savedPipeline =
+          (Map<String, Object>) objectMapper.readValue(response.getBody().in(), Map.class);
+      outputs.put("pipeline.id", savedPipeline.get("id"));
+    } catch (Exception e) {
+      log.error("Unable to deserialize saved pipeline, reason: ", e.getMessage());
```

Comment:
Can you log the exception here? It can be hard to track down errors just from the message. 

```
       log.error("Unable to deserialize saved pipeline", e);
```

should be enough as slf4j should include the stack trace with the right format.

Reply:
Yes...will add that